bindings/src/main/java/io/pravega/storage/hdfs/HDFSChunkStorage.java

```diff
@@ -0,0 +1,360 @@
+     * Creates a new instance of the HDFSStorage class.
+     *
+     * @param config The configuration to use.
+     */
+    HDFSChunkStorage(HDFSStorageConfig config) {
+        Preconditions.checkNotNull(config, "config");
+        this.config = config;
+        this.closed = new AtomicBoolean(false);
+        initialize();
+    }
+
+    //endregion
+
+    //region capabilities
+
+    @Override
+    public boolean supportsConcat() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsAppend() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsTruncation() {
+        return false;
+    }
+
+    //endregion
+
+    //region AutoCloseable Implementation
+
+    @Override
+    public void close() {
+        if (!this.closed.getAndSet(true)) {
+            if (this.fileSystem != null) {
+                try {
+                    this.fileSystem.close();
+                    this.fileSystem = null;
+                } catch (IOException e) {
+                    log.warn("Could not close the HDFS filesystem: {}.", e);
+                }
+            }
+        }
+    }
+
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        ensureInitializedAndNotClosed();
+        try {
+            FileStatus last = fileSystem.getFileStatus(getFilePath(chunkName));
+            ChunkInfo result = ChunkInfo.builder().name(chunkName).length(last.getLen()).build();
+            return result;
+        } catch (IOException e) {
+            throwException(chunkName, "doGetInfo", e);
+        }
+        return null;
+    }
+
+    @Override
+    protected ChunkHandle doCreate(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        ensureInitializedAndNotClosed();
+        try {
+            Path fullPath = getFilePath(chunkName);
+            // Create the file, and then immediately close the returned OutputStream, so that HDFS may properly create the file.
+            this.fileSystem.create(fullPath, READWRITE_PERMISSION, false, 0, this.config.getReplication(),
+                    this.config.getBlockSize(), null).close();
+            log.debug("Created '{}'.", fullPath);
+
+            // return handle
+            return ChunkHandle.writeHandle(chunkName);
+        } catch (FileAlreadyExistsException e) {
+            throw new ChunkAlreadyExistsException(chunkName, "HDFSChunkStorage::doCreate");
+        } catch (IOException e) {
+            throwException(chunkName, "doCreate", e);
+        }
+        return null;
+    }
+
+    @Override
+    protected boolean checkExists(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        ensureInitializedAndNotClosed();
+        FileStatus status = null;
+        try {
+            status = fileSystem.getFileStatus(getFilePath(chunkName));
+        } catch (IOException e) {
+            // HDFS could not find the file. Returning false.
```

Comment:
Why is it swallowing the exception? Maybe the file is there and the exception is due to something else, in which case the return value is incorrect.

Reply:
This code is copied from the existing `HDFSStorage.java`. However the code within the `try` block is also very different. Can we check the contracts for that method to see if it warrants this whole try-catch block in here too?

Reply:
fixed.